app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -132,9 +137,49 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
       }
     }
 
-    startServerButton.setOnClickListener { startStopServer() }
+    startServerButton.setOnClickListener {
+      if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P) {
+        if (checkCoarseLocationAccessPermission()) {
+          startStopServer()
+        }
+      } else {
```

Comment:
so when we click this button on any api level below p it will never start the server?

Reply:
@macgills I don't understand the question. If it's greater than p then it will check permission then `startStopServer` otherwise it'll just `startStopServer`.

Reply:
Ah my apologies, `startStopServer()` this irritatingly named method made my eyes glaze over and I read it as separate methods `start/stop` in different locations.

Okay, we can still make this better.

```
if(Build.VERSION.SDK_INT <= Build.VERSION_CODES.P || checkCoarseLocationAccessPermission()){
startStopServer()
}
```
thats all you need to say

Reply:
wow! so clean.